docs/proposals/metrics-plumbing.md

```diff
@@ -0,0 +1,130 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/resource-qos.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Resource Usage Metrics plumbing in Kubernetes
+
+**Author**: Vishnu Kannan (@vishh)
+
+**Status**: Draft proposal; some parts are already implemented
+
+* This document presents a design for handling container metrics in Kubernetes clusters*
+
+## Motivation
+
+Resource usage metrics are critical for various reasons:
+* Monitor and maintain the health of the cluster and user applications.
+* Improve the efficiency of the cluster by making more optimal scheduling decisions and enabling components like auto-scalers.
+
+There are multiple types of metrics that describe the state of a container.
+Numerous strategies exist to aggregate these metrics from containers.
+There are a variety of storage backends that can handle metrics.
+
+This document presents a design to abstract out collection and storage backends, and provide stable Kubernetes APIs that can be consumed by users and other cluster components.
+
+## Introduction
+
+Container metrics can be of two types.
+
+1. `Compute resource metrics` refers to compute resources being used by a container. Ex.: CPU, Memory, Network, File-system
+2. `Service metrics` refers to container app specific metrics. Ex: QPS, query latency, etc.
```

Comment:
[cAdvisor](https://github.com/google/cadvisor/tree/master/collector) to begin with. We can extend support for collectd or other collectors if necessary, but we need to somehow whitelist the metrics that matter to the rest of the system (auto-scaling mainly).